Automattic/vip-go-mu-plugins
# Healthcheck that fully loads customer code and plugins before determining whether a site is healthy or not.

## Description

This should help us to determine whether a site is healthy or not (we can use it as a readiness probe and to potentially power other places).

The main idea here is to perform the health check at a point in the request lifecycle where all the code is loaded (as opposed to `cache-healthcheck?` which gets executed early on.


## Changelog Description

### Plugin Introduced: Healthcheck

We've introduced a new Healthcheck which runs at a point where all of the site's codebase should be loaded. Any errors at this point could be used as a strong signal that the application is unhealthy.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

- Check out PR.
- Hit `/vip-healthcheck` in the browser or via curl:

```
 curl --head http://mu.vipdev.lndo.site/vip-healthcheck
HTTP/1.1 200 OK
Cache-Control: no-cache, must-revalidate, max-age=0
Content-Type: text/plain;charset=UTF-8
Date: Wed, 27 Apr 2022 22:35:29 GMT
Expires: Wed, 11 Jan 1984 05:00:00 GMT
```

- It should display return an empty document with `200` status code.
- Now add an error via `$hc->add_error( 'test error'  )` and hit the URL again, it should display the error text and return a 500 error.
